packages/core/tocco-ui/src/EditableValue/editors/useTimeEdit.js

```diff
@@ -0,0 +1,130 @@
+import {format, isMatch, parse} from 'date-fns'
+import {useRef, useState, useEffect} from 'react'
+import {react} from 'tocco-util'
+
+const TimeInitialFormat = 'HH:mm:ss'
+const TimeFormat = 'HH:mm'
+const OneHourFormat = 'H'
+const HoursFormat = 'HH:'
+
+// e.g. 1, 12, 12:, 12:3, 12:34, 1:34
+const isIntermediateValid = val => /^(\d{0,2}|\d{1,2}:\d{0,2})$/.test(val)
+const isValid = val => isMatch(val, TimeFormat)
+
+const parseValueToDatetime = val => {
+  if (isMatch(val, TimeInitialFormat)) {
+    return parse(val, TimeInitialFormat, new Date())
+  }
+
+  if (isMatch(val, TimeFormat)) {
+    return parse(val, TimeFormat, new Date())
+  }
+
+  if (isMatch(val, HoursFormat)) {
+    return parse(val, HoursFormat, new Date())
+  }
+  if (isMatch(val, OneHourFormat)) {
+    return parse(val, OneHourFormat, new Date())
+  }
+
+  return null
+}
+const formatDatetime = datetime => format(datetime, TimeFormat)
+const formatValue = val => {
+  const datetime = parseValueToDatetime(val)
+  return datetime && !isNaN(datetime) ? formatDatetime(datetime) : ''
+}
+
+const toggleColon = (value, prevValue) => {
+  if (!value || !prevValue) {
+    return value
+  }
+
+  if (value.length === 2 && prevValue.length !== 3 && value.indexOf(':') === -1) {
```

Comment:
(I personally would prefer `includes` to `indexOf` to check if a string includes a character)